Version2 of py_dss_interface

temcdrm opened the pull request:
- using the more pythonic py_dss_interface v2
- progress bar appears in DAILY solution mode
- input bus order is maintained in Network.json files, which are now **directed graphs** in networkx
- smoothed the daily loadshapes (see https://github.com/pnnl/i2x/tree/version1/test), to mitigate OpenDSS piecewise constant interpolation
- workaround for adverse interactions between solar inverter control functions, and a synchronous generator in PV mode (see
[GeneratorPVTests.pdf](https://github.com/pnnl/i2x/files/11653544/GeneratorPVTests.pdf)
)
- for https://github.com/pnnl/i2x/issues/7, clarified that min and max node voltages only consider the last load flow solution. Use the min and max PV voltages for evaluation; they are based on monitors and consider all time points.
- hca.py runs but the answers are unverified

Commit:
working on bus order, GUI, and py-dss-interface version updates

Commit:
code runs with opendssdirect v2, but the event logs are empty

Commit:
writing the event log works now

Commit:
working on inverter control fixes

Commit:
added smooth loadshapes by interpolants on 1-second intervals

Commit:
require OpenDSSDirect with progress bar and control function updates

Commit:
workaround for generator in PV mode; updated inverter control examples for smoother loadshapes

Commit:
keep the original steamgen1 settings as a comment line

Commit:
revert to using the system.event_log function, which has been fixed in py_dss_interface

Commit:
clarifying the interpretation of min and max node voltage

eranschweitzer commented:
@temcdrm  a few question:
* Could you clarify the change to the IEEE9500 model here:
https://github.com/pnnl/i2x/blob/2478b8729af8afc0261d8dc150e24eaedd7d1de2/src/i2x/models/ieee9500/HCABase.dss#L67-L69

* I see that in the IEEE 9500 model the `set eventlogdefault=yes` is commented out: https://github.com/pnnl/i2x/blob/2478b8729af8afc0261d8dc150e24eaedd7d1de2/src/i2x/models/ieee9500/HCABase.dss#L2
On the other hand in the IEEE-LVN network model this is active:
https://github.com/pnnl/i2x/blob/2478b8729af8afc0261d8dc150e24eaedd7d1de2/src/i2x/models/ieee_lvn/SecPar.dss#L3

* out of curiosity, what is the purpose of the [`clean.bat`](https://github.com/pnnl/i2x/blob/2478b8729af8afc0261d8dc150e24eaedd7d1de2/clean.bat) file

* How come we are using the quadratic interpolation rather than the linear one as default? it appears a bit "wavy" and also has the somewhat peculiar side effect of going above 1 pu.

* What is the meaning of the non-unity power factor for the inverter controls? I thought the whole idea of an automatic control is that the power factor is not fixed... Also, what is negative vs. positive? that is, which one consumes vars vs. injects vars?

* I did not verify the new graph model line by line, but the plots etc. seem to suggest it was converted correctly.

* Everything also runs on my end, however, I noticed that the number of Large DER sometimes varies. Is it possible for the residential pv (e.g. 20% of load buses) to create something that is > 100 kW? If so, do we really want that? it seems like that is not necessarily realistic for a residential installation, no?

Commit:
restore the 9500-node event log

temcdrm commented:
- It was described in the GeneratorPVTests.pdf file attached to the first comment. In line 3 here https://github.com/pnnl/i2x/blob/develop/src/i2x/models/ieee9500/Generators.dss, the steam generator was set to model 3, a PV bus
- That comment was left in while debugging the generator issue, now corrected.
- It removes output files created while using standalone OpenDSS for debugging models, before git commit or push
- I want the default load variations to be as smooth as possible, which helps isolate the effect of PV inverters.  A user can choose linear or cubic if desired.
- Fixed power factor is an inverter control mode required in IEEE 1547-2018, and some utilities use it to mitigate voltage issues without exposing themselves to the perceived risk of autonomous control action.  Negative fixed power factor means the inverter absorbs vars while generator watts.
- Good
- Possibly.  Please record a bus number the next time you notice this.  The algorithm tries to size PV close to the companion load size, so a large single-phase load at 208-V base could have a 100-kW solar installation created.  That would represent an aggregate house(s), not just one.  The load is really sized for the service transformer.  When you have a bus number, please create a separate issue.
